### PR TITLE
Rename FSE subfeature to dotcom fse

### PR DESCRIPTION
**Pull request by noahtallen**
#### Changes proposed in this Pull Request 
* Change `full-site-editing-plugin/full-site-editing` to `full-site-editing-plugin/dotcom-fse`

@vindl and I chatted today about the plugin rename, and at least for now, we feel it'll introduce even greater confusion if we rename the plugin. Instead, for now, we'll improve clarity in the codebase by changing the `full-site-editing` subfeature to `dotcom-fse`. This will help us especially when we begin to incorporate core parts of FSE to dotcom.

### Testing instructions
Essentially, just make sure nothing changed with the build. Everything should behave exactly as it did before.

1. Local build should work correctly in a local WordPress install and on dotcom
2. Fse build step in CircleCI should complete and the artifact should work correctly on the sandbox

### CODE REVIEW

**Comment by vindl** [comment]
I think it still makes sense to do this, although some conflicts have appeared in the meantime that we need to resolve before merging it.

**Comment by noahtallen** [comment]
@vindl rebased and ready to test again. Also fixed some issues with the asset include files

File: apps/full-site-editing/full-site-editing-plugin/dotcom-fse/class-full-site-editing.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Full_Site_Editing {
 	 *
 	 * @var array
 	 */
-	private $template_post_types = [ 'wp_template_part' ];
+	private $template_post_types = array( 'wp_template_part' );
```
Comment (noahtallen) [inline apps/full-site-editing/full-site-editing-plugin/dotcom-fse/class-full-site-editing.php]: uhhhhhh phpcs, what you doing here?

Comment (vindl) [reply apps/full-site-editing/full-site-editing-plugin/dotcom-fse/class-full-site-editing.php]: Back in the day I was trying to make it consistent with shorthand syntax, but if `phpcs` wants `array` by default I'm fine with that, as long as we don't have to fight it on each PR. 😄

Comment (noahtallen) [reply apps/full-site-editing/full-site-editing-plugin/dotcom-fse/class-full-site-editing.php]: Definitely. Ideally, we'd let phpcs allow `[  ]`, but probably not worth it for this PR

**Pull request merged**